operator/controllers/btpoperator_controller_test.go

```diff
@@ -46,102 +50,123 @@ func newFakeK8s(c client.Client) *fakeK8s {
 }
 
 func (f *fakeK8s) DeleteAllOf(ctx context.Context, obj client.Object, opts ...client.DeleteAllOfOption) error {
```

Comment:
This fake deleteAllOf does not work as expected. Test with hardDelete timeout doesnt work, cause SI and SB are moeved wothout any errors. Why, because Btp operator Controller is not properly installed (only deployment) and SI and SB do not have finalizers.  

We need to redesign how we want to test hardDelete, softdelete after timeout, and soft delete after hardDelete failure.

Reply:
I created 2 fake k8s clients with shadowed `DeleteAllOf` method to fit our test cases. These clients are used in deprovisioning tests. Also I added func to apply finalizers to SIs and SBs created in deprovisioning tests